nilearn/reporting/html_report.py

```diff
@@ -284,17 +282,19 @@ def generate_report(estimator):
     # Generate a unique ID for this report
     data["unique_id"] = str(uuid.uuid4()).replace("-", "")
 
+    if data.get("title") is None:
+        data["title"] = estimator.__class__.__name__
+
     warning_messages = []
 
     if estimator.reports is False:
         warning_messages.append(
             "\nReport generation not enabled!\nNo visual outputs created."
         )
 
-    if (
-        not hasattr(estimator, "_reporting_data")
-        or not estimator._reporting_data
-    ):
+    try:
+        check_is_fitted(estimator)
+    except NotFittedError:
```

Comment:
```suggestion
    if not estimator.__sklearn_is_fitted__()
```

simpler than a try catch and all estimators MUST have this method